torch/testing/_internal/distributed/rpc/dist_optimizer_test.py

```diff
@@ -18,11 +18,13 @@ class MyModule:
     lock = threading.Lock()
 
     def __init__(self):
-        # avoid race where 2 modules could be initialized
-        # concurrently thus changing the order random numbers are drawn.
-        with MyModule.lock:
-            torch.manual_seed(0)
-            self.w = torch.rand((3, 3), requires_grad=True)
+        # cannot directly use torch.manual_seed(0) as all threads share the same
```

Comment:
Am I right that the previous version with the lock was working fine but you changed it to be consistent with the other tests?

Reply:
No, the previous version was flaky because the RNG is not only used here, but also in testy body of `test_dist_optim`. What could happen is that rank 0 sending rpc to rank 1, which triggers module construction here in an RPC thread, but the main testing thread in `test_dist_optim` (look for `t1` and `t2` creation) might also draw from the same default RNG instance concurrently. As a result, we cannot guarantee the order.